tracer/src/Datadog.Trace/OTelMetrics/OtlpExporter.cs

```diff
@@ -0,0 +1,275 @@
+        {
+            if (metrics.Count == 0)
+            {
+                Log.Debug("No metrics to export.");
+                return ExportResult.Success; // Success - nothing to do
+            }
+
+            // Record export attempt
+            TelemetryFactory.Metrics.RecordCountMetricsExportAttempts(GetProtocolTag());
+
+            try
+            {
+                // Take snapshots of all metrics for export
+                var metricPoints = new List<MetricPoint>();
+                foreach (var metric in metrics)
+                {
+                    metric.TakeSnapshot(outputDelta: _settings.OtlpMetricsTemporalityPreference == Configuration.OtlpTemporality.Delta);
+                    metricPoints.Add(metric);
+                }
+
+                // Serialize to OTLP protobuf format
+                var otlpPayload = OtlpMetricsSerializer.SerializeMetrics(metricPoints, _settings);
+
+                // Send to OTLP endpoint asynchronously for optimal performance
+                var success = await SendOtlpRequest(otlpPayload).ConfigureAwait(false);
+
+                if (success)
+                {
+                    // Record successful export
+                    TelemetryFactory.Metrics.RecordCountMetricsExportSuccesses(GetProtocolTag());
+                    return ExportResult.Success;
+                }
+                else
+                {
+                    // Record failed export
+                    TelemetryFactory.Metrics.RecordCountMetricsExportFailures(GetProtocolTag());
+                    return ExportResult.Failure;
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error exporting OTLP metrics.");
+                // Record failed export
+                TelemetryFactory.Metrics.RecordCountMetricsExportFailures(GetProtocolTag());
+                return ExportResult.Failure;
+            }
+        }
+
+        /// <summary>
+        /// Shuts down the exporter and ensures all pending exports complete.
+        /// </summary>
+        /// <param name="timeoutMilliseconds">Maximum time to wait for shutdown</param>
+        /// <returns>True if shutdown completed successfully, false otherwise</returns>
+        public override bool Shutdown(int timeoutMilliseconds)
+        {
+            try
+            {
+                using var cts = new CancellationTokenSource(timeoutMilliseconds);
+                _httpClient.Dispose();
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Log.Warning(ex, "Error during OTLP exporter shutdown");
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the protocol tag for telemetry metrics based on the configured OTLP protocol.
+        /// </summary>
+        private Telemetry.Metrics.MetricTags.Protocol GetProtocolTag()
```

Comment:
super nit: You can calculate this string during the constructor so it's just a field access rather than calling a method